plugins/synapticsmst/synapticsmst-device.c

```diff
@@ -586,139 +1094,242 @@ synapticsmst_device_write_firmware (SynapticsMSTDevice *device,
 				     "configuration checksum error");
 		return FALSE;
 	}
-
+	/* Firmware Size */
 	checksum = 0;
 	offset = 0x400;
-	code_size = (*(payload_data + offset) << 8) + *(payload_data + offset + 1);
-	if (code_size >= 0xFFFF) {
-		g_set_error_literal (error,
-				     G_IO_ERROR,
-				     G_IO_ERROR_INVALID_DATA,
-				     "invalid firmware size");
-		return FALSE;
+	if(isChipPanamera == TRUE){
+		code_size = (*(payload_data + offset) << 24);
+		code_size += (*(payload_data + offset + 1)<<16);
+		code_size += (*(payload_data + offset + 2)<<8);
+		code_size += (*(payload_data + offset + 3));
+		
+		if (code_size >= 0x02FFFF) {
+			g_set_error_literal (error,
+						G_IO_ERROR,
+						G_IO_ERROR_INVALID_DATA,
+						"invalid firmware size");
+			return FALSE;
+		}
 	}
-	for (guint32 i = 0; i < (code_size + 17); i++)
-		checksum += *(payload_data + offset + i);
-
-	if (checksum & 0xFF) {
-		g_set_error_literal (error,
-				     G_IO_ERROR,
-				     G_IO_ERROR_INVALID_DATA,
-				     "firmware checksum error");
-		return FALSE;
+	else{
+		code_size = (*(payload_data + offset) << 8) + *(payload_data + offset + 1);
+		if (code_size >= 0xFFFF) {
+			g_set_error_literal (error,
+						G_IO_ERROR,
```

Comment:
This whitespace got a little messed up in the process of special casing panamera.

Reply:
Yes, the format is changed. What do you mean exactly?

Reply:
`G_IO_ERROR` doesn't align to the `error` on line before it anymore.